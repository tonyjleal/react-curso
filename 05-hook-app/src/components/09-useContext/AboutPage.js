import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const AboutPage = () => {

    const { user, setUser } = useContext(UserContext);


    const handleClick = () => {
        setUser({});
    }

    return <div>
        <h1>About Page</h1>
        <hr />
        <pre>
            {
                JSON.stringify(user, null, 3)
            }
        </pre>

        <button
            className='btn btn-warning'
            onClick={handleClick}
        >
            Logout
        </button>
    </div>;
};
