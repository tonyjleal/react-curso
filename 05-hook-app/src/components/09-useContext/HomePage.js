import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const HomePage = () => {

  const userContext = useContext(UserContext);

  console.log(userContext);

  return <div>
      <h1>Home Page</h1>
      <hr />
  </div>;
};
