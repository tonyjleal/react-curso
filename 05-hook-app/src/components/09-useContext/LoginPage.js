import React from 'react';

export const LoginPage = () => {
    return <div>
        <h1>Again Page</h1>
        <hr />
    </div>;
};
