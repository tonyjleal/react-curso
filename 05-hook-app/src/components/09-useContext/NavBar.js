import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
  return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">
                
                <Link className="navbar-brand" to={'/'}>useContext</Link>

                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink className={`nav-link ${(navData) => navData.isActive && 'active' }`} aria-current="page" to={'/'}>Home</NavLink>
                    <NavLink className={`nav-link ${(navData) => navData.isActive && 'active' }`} to={'/about'}>About</NavLink>
                    <NavLink className={`nav-link ${(navData) => navData.isActive && 'active' }`} to={'/login'}>Login</NavLink>
                </div>
                </div>
            </div>
        </nav>
  );
};
