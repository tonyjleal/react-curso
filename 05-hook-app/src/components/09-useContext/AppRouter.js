import React from 'react';
import {
    BrowserRouter as Router,
    Route,
    Routes,
    Navigate
  } from "react-router-dom";
import { AboutPage } from './AboutPage';
import { HomePage } from './HomePage';
import { LoginPage } from './LoginPage';
import { NavBar } from './NavBar';


export const AppRouter = () => {
  return <Router>
        <div>
            <NavBar />
            
            <Routes>
                <Route path="/"      element={ <HomePage />  } />
                <Route path="/about" element={ <AboutPage /> } />
                <Route path="/login" element={ <LoginPage /> } />

                <Route path="*"  element={<Navigate to="/" />} />
            </Routes>
        </div>
    </Router>
  ;
};
