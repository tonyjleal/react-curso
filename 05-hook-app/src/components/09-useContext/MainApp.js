import React from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export const MainApp = () => {

    const user = {
        id: 123,
        name: 'Test',
        email: 'test@test.com'
    }

    return (
        <UserContext.Provider value={ user }>
            <AppRouter />
        </UserContext.Provider>
    )
};
