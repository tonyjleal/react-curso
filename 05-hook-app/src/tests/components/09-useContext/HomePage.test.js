import { shallow, mount } from 'enzyme'
import { HomePage } from '../../../components/09-useContext/HomePage'
import { UserContext } from '../../../components/09-useContext/UserContext'

describe('Test Component <HomePage />', () => { 
    
    const user = {
        name: 'Antonio',
        email: 'test@test.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{user}}>
            <HomePage />
        </UserContext.Provider>
    )


    test('should show component success', () => { 
        
        expect(wrapper).toMatchSnapshot();

     })

})