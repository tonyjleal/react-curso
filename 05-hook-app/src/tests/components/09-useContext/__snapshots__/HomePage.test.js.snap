// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component <HomePage /> should show component success 1`] = `
<div>
  <h1>
    Home Page
  </h1>
  <hr />
  <pre>
    {
   "name": "Antonio",
   "email": "test@test.com"
}
  </pre>
</div>
`;
